lib/iris/tests/unit/common/metadata/test_CubeMetadata.py

```diff
@@ -95,15 +96,349 @@ def test_bases(self):
+    set to `False` if the secondary inputs do not match.
+
+    Calling with different values for the keywords aims to show that the main operation
+    has the expected value, from _ALL_RESULTS, the ***same in essentially all cases***
+    ( though modified in specific ways for some factors ).
+
+    This regularity also demonstrates the required independence over the other
+    test-factors, i.e. global/local attribute type, and right-left order.
+    """
+    # Just for comfort, check that inputs are all one of a few single characters.
+    assert all(
+        (item in list("ABCDX")) for item in (primary_inputs + secondary_inputs)
+    )
+    # Interpret "primary" and "secondary" inputs as "global" and "local" attributes.
+    if check_global_not_local:
+        global_values, local_values = primary_inputs, secondary_inputs
+    else:
+        local_values, global_values = primary_inputs, secondary_inputs
+
+    # Form 2 inputs to the operation :  Make left+right split-attribute input
+    # dictionaries, with both the primary and secondary attribute value settings.
+    input_dicts = [
+        CubeAttrsDict(
+            globals=make_attrsdict(global_value),
+            locals=make_attrsdict(local_value),
+        )
+        for global_value, local_value in zip(global_values, local_values)
+    ]
+    # Make left+right CubeMetadata with those attributes, other fields all blank.
+    input_l, input_r = [
+        CubeMetadata(
+            **{
+                field: attrs if field == "attributes" else None
+                for field in CubeMetadata._fields
+            }
+        )
+        for attrs in input_dicts
+    ]
+
+    if check_reversed:
+        # Swap the inputs to perform a 'reversed' calculation.
+        input_l, input_r = input_r, input_l
+
+    # Run the actual operation
+    result = getattr(input_l, operation_name)(
+        input_r, lenient=check_is_lenient
+    )
+
+    if operation_name == "difference" and check_reversed:
+        # Adjust the result of a "reversed" operation to the 'normal' way round.
+        # ( N.B. only "difference" results are affected by reversal. )
+        if isinstance(result, CubeMetadata):
+            result = result._replace(attributes=result.attributes[::-1])
+
+    # Extract, from the operation result, the value to be tested against "expected".
+    result = extract_result_value(result, check_global_not_local)
+
+    # Get the *expected* result for this operation.
+    which = "lenient" if check_is_lenient else "strict"
+    primary_key = "primary" + primary_inputs
+    expected = _ALL_RESULTS[operation_name][primary_key][which]
+    if operation_name == "equal" and expected:
+        # Account for the equality cases made `False` by mismatched secondary values.
+        secondary_1, secondary_2 = secondary_inputs
+        if (
+            secondary_1 != "X"
+            and secondary_2 != "X"
+            and secondary_1 != secondary_2
+        ):
+            expected = False
```

Comment:
Doesn't this need some nuance depending on whether we are testing lenient or strict?

Reply:
Hmm that's interesting...  Really good spot !
So, "primaryAA" plus "secondaryCX" would show this, being True if lenient and False if strict.
But I think this is missed because I am only doing the secondary options XX / CC / CD

So, if I extend this by adding a "secondaryCX" option, it does indeed fail, 
In `Test___eq__.test_splitattrs_global_local_independence[strict-primaryAA-secondaryCX]`,
the actual results are lenient=True, strict=False, but the "expected" was True in both cases.

I think that means that the above test needs more "nuance", depending on strict/lenient as you suggest.
I will look into it ...

So I do also think this makes a case for adding more 'secondary' options.  I'll think on.

Reply:
Ok here's my fix : https://github.com/SciTools/iris/pull/5538/commits/dca5dce1d0e3d795329b819fb3d40d965752bb66

I also expanded the "secondary" test options, so we are now checking cases where a missing secondary value does or does not align with a missing primary value. 
I can confirm that some of those new cases _fail_ without the logic fix.